libs/langchain-community/src/retrievers/matryoshka_retrieval.ts

```diff
@@ -0,0 +1,128 @@
+import { CallbackManagerForRetrieverRun } from "@langchain/core/callbacks/manager";
```

Comment:
I think this should live in `langchain` and not community since it's more of an algorithm?